### ISSUE TITLE
Undo button in snackbar.

### ISSUE THREAD

**Issue by Akshatji800** [CCExtractor/Flood_Mobile#113]
**Is your feature request related to a problem? Please describe.**
There is no way to undo our actions performed. For example suppose a user by mistake deletes a torrent download so he will not be any chance to undo his action. Instead he should be given a chance for some seconds to do it.

**Describe the solution you'd like**
Undo button can be implemented in the snackbar instead of dismiss button to undo the action performed.

**Comment by BhuvanAde** [comment]
@Akshatji800  Can I work on this issue?

**Comment by Akshatji800** [comment]
Sure @BhuvanAde assigning the issue to you.

**Comment by amitamrutiya2210** [comment]
hey @Akshatji800 if @BhuvanAde is not working on this issue then i want to work.

**Comment by Akshatji800** [comment]
@BhuvanAde is there any progress on this issue?

**Comment by BhuvanAde** [comment]
I was facing a issue; I think it is solved now. I will soon send the PR.

**Comment by A-Kashif108** [comment]
As @BhuvanAde unassigned himself. Can I please take this issue?

**Comment by Akshatji800** [comment]
Sure @A-Kashif108 assigning this issue to you

**Comment by Interstellar07** [comment]
Is anyone working on this? If not kindly assign this to me. I would love to contribute

**Comment by Sudhanva-Nadiger** [comment]
Can I Work on this @Akshatji800 ?

**Comment by Skdahale** [comment]
@Akshatji800  Can I work on this issue?

**Comment by ALIRAZA17** [comment]
@Akshatji800 Is anyone working on this one. If not Can you assign me?

**Comment by lalitkumawat1m** [comment]
can I work on this issue ?

**Comment by Akshatji800** [comment]
> can I work on this issue ?

Sure @lalitkumawat1m I am assigning this issue to you.

**Comment by lalitkumawat1m** [comment]
Dear @Akshatji800,

I am writing to you today because I am having an issue running the Flood mobile app locally. When I try to run the app, I get the following error:

Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'
I have tried the following to fix the error:

I have restarted my computer.
I have closed and reopened Visual Studio Code.
I have restarted the Dart Analysis Server.
However, the error still persists.

<img width="960" alt="Screenshot 2023-07-14 205156" src="https://github.com/CCExtractor/Flood_Mobile/assets/91591901/7e747f7d-b19e-4828-a4d1-d07d6f81346d">

I would be grateful if you could help me to resolve this issue.

Thank you for your time and attention to this matter.

I hope this helps! Let me know if you have any other questions.

**Comment by Akshatji800** [comment]
> Dear @Akshatji800,
> 
> I am writing to you today because I am having an issue running the Flood mobile app locally. When I try to run the app, I get the following error:
> 
> Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/app_localizations.dart' I have tried the following to fix the error:
> 
> I have restarted my computer. I have closed and reopened Visual Studio Code. I have restarted the Dart Analysis Server. However, the error still persists.
> 
> <img alt="Screenshot 2023-07-14 205156" width="960" src="https://user-images.githubusercontent.com/91591901/253612414-7e747f7d-b19e-4828-a4d1-d07d6f81346d.png">
> I would be grateful if you could help me to resolve this issue.
> 
> Thank you for your time and attention to this matter.
> 
> I hope this helps! Let me know if you have any other questions.

Hello @lalitkumawat1m please go through the Contribution.md file and follow the setup instructions. This will surely solve your issue. There is just a command that you need to run before running the `flutter run` command to resolve the issue. I want you to explore Contribution.md file and find that command yourself instead of me helping you directly. :)

**Comment by tb-rules10** [comment]
@Akshatji800 can you assign me this issue?

I am considering implementing an 'Undo' button specifically for the delete functionality. As for other snackbars, the dismiss button will remain.  Would that be fine?

**Comment by Akshatji800** [comment]
> @Akshatji800 can you assign me this issue?
> 
> I am considering implementing an 'Undo' button specifically for the delete functionality. As for other snackbars, the dismiss button will remain. Would that be fine?

@tb-rules10 the dismiss button will remain for each and every snackbar (even for the delete one), you need to implement an undo button beside the dismiss button. Hope that is clear. Once you confirm that you understood the issue then please reply to this comment and I will assign you the issue.

**Comment by tb-rules10** [comment]
Yes, I understood it. You can assign it to me. Just one more thing, undo button has to be implemented only for the delete bar if I'm right?

**Comment by Akshatji800** [comment]
> Yes, I understood it. You can assign it to me. Just one more thing, undo button has to be implemented only for the delete bar if I'm right?

for now you start with implementing it for delete feature, later we can go ahead with other features as well if it look logical to implement an undo feature for them. Anyways I am assigning this issue to you.

**Comment by tb-rules10** [comment]
Any update on my pr?
